src/main/java/in/org/projecteka/hdaf/HdafConfiguration.java

```diff
@@ -59,6 +60,21 @@ public DiscoveryRepository discoveryRepository(PgPool pgPool) {
         return new DiscoveryRepository(pgPool);
     }
 
+    @Bean
+    public LinkRepository linkRepository(DbOptions dbOptions) {
+        PgConnectOptions connectOptions = new PgConnectOptions()
+                .setPort(dbOptions.getPort())
+                .setHost(dbOptions.getHost())
+                .setDatabase(dbOptions.getSchema())
+                .setUser(dbOptions.getUser())
+                .setPassword(dbOptions.getPassword());
+
+        PoolOptions poolOptions = new PoolOptions()
+                .setMaxSize(dbOptions.getPoolSize());
+
+        return new LinkRepository(PgPool.pool(connectOptions, poolOptions));
```

Comment:
Can you use PgPool bean that is already created?

Reply:
sure